go-runtime/ftl/reflection/type_registry.go

```diff
@@ -66,10 +50,21 @@ func NewTypeRegistry(options ...TypeRegistryOption) *TypeRegistry {
 	return t
 }
 
+// Register applies all the provided options to the singleton TypeRegistry
+func Register(options ...TypeRegistryOption) {
+	for _, o := range options {
+		o(singletonTypeRegistry)
+	}
+}
+
 // RegisterSumType registers a Go sum type with the type registry.
 //
 // Sum types are represented as enums in the FTL schema.
-func (t *TypeRegistry) RegisterSumType(discriminator reflect.Type, variants map[string]reflect.Type) {
+func RegisterSumType(discriminator reflect.Type, variants map[string]reflect.Type) {
```

Comment:
I think this is unused now, we can probably get rid of it?

Reply:
Looks like it was used in encoding_test.go for getting around the package validation in reflection.go, but that's kind of gross testing using test-only logic, so I removed this and updated the tests to use `reflection.WithSumType` with `reflection.AllowAnyPackageForTesting`. We already had `reflection.allowAnyPackageForTesting` for reflection_test and type_registry_test, so I just made that variable public. It's gross but less gross than sidestepping `WithSumType`, I think, but please let me know if you think of something better. Thanks for catching this!